kubernetes-model-generator/kubernetes-model-core/src/main/java/io/fabric8/kubernetes/api/model/HasMetadata.java

```diff
@@ -232,4 +235,109 @@ default boolean isFinalizerValid(String finalizer) {
   default boolean removeFinalizer(String finalizer) {
     return getMetadata().getFinalizers().remove(finalizer);
   }
+
+  default boolean hasOwnerReferenceFor(HasMetadata owner) {
+    return getOwnerReferenceFor(owner).isPresent();
+  }
+
+  default boolean hasOwnerReferenceFor(String ownerUid) {
+    return getOwnerReferenceFor(ownerUid).isPresent();
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(HasMetadata owner) {
+    if (owner == null) {
+      return Optional.empty();
+    }
+
+    final String ownerUID = owner.getMetadata().getUid();
+    return getOwnerReferenceFor(ownerUID);
+  }
+
+  default Optional<OwnerReference> getOwnerReferenceFor(String ownerUid) {
+    if (ownerUid == null || ownerUid.isEmpty()) {
+      return Optional.empty();
+    }
+
+    return getMetadata().getOwnerReferences().stream()
+      .filter(o -> ownerUid.equals(o.getUid()))
+      .findFirst();
+  }
+
+  default OwnerReference addOwnerReference(HasMetadata owner) {
+    if (owner == null) {
+      throw new IllegalArgumentException("Cannot add a reference to a null owner to '" + getMetadata().getName() + "' " + getKind());
+    }
+
+    final ObjectMeta metadata = owner.getMetadata();
+    final OwnerReference ownerReference = new OwnerReferenceBuilder()
+      .withUid(metadata.getUid())
+      .withApiVersion(owner.getApiVersion())
+      .withName(metadata.getName())
+      .withKind(owner.getKind())
+      .build();
+    return addOwnerReference(ownerReference);
+  }
+
+  default OwnerReference addOwnerReference(OwnerReference ownerReference) {
```

Comment:
It seems like there are two (possibly static) methods that could be created:

createOwnerReference(HasMetadata)
validateOwnerReference(OwnerReference)

I'm honestly not sure if addOwnerReference needs to perform validation - does kube not preform these validations?  And wouldn't the user already be using addOwnerReference(HasMetadata)?  And couldn't the object be modified after this?